helix-term/src/commands/lsp.rs

```diff
@@ -904,7 +904,12 @@ pub fn signature_help_impl(cx: &mut Context, invoked: SignatureHelpInvoked) {
                         Some((start, start + string.len()))
                     }
                     lsp::ParameterLabel::LabelOffsets([start, end]) => {
-                        Some((*start as usize, *end as usize))
+                        // LS sends offsets based on utf-16 based string representation
```

Comment:
Oh if these are utf16 then you should use the utf16_to_byte_idx method instead

Reply:
I had to add `str_indices` as an explicit dependency since `ropey::str_utils` doesn't expose the utf-16 methods. On a separate note, rust-analyzer does a [simple `.chars().count()`][1] to send us the utf-16 offsets, so wouldn't their implementation trip up on `char`s that take more than 16 bits, i.e. when `char.len_utf16() > 1` ?

For a contrived example:

```rust
let len = '💣'.len_utf16();
assert_eq!(len, 2);

let len = "💣".chars().count();
assert_eq!(len, 1);
```

[1]: https://github.com/rust-lang/rust-analyzer/blob/0b131bc78eece8be69b5d3633f4db04cf2c1151b/crates/rust-analyzer/src/to_proto.rs#L368-L369
[2]: https://doc.rust-lang.org/std/primitive.char.html#method.len_utf16